modules/danmarketplaceBidAdapter.js

```diff
@@ -0,0 +1,144 @@
+};
+
+/**
+ * Dentsu Aegis Network Marketplace Bid Adapter.
+ * Contact: niels@baarsma.net
+ *
+ */
+export const spec = {
+  code: BIDDER_CODE,
+
+  aliases: ['DANMarketplace', 'DAN_Marketplace'],
+
+  isBidRequestValid: function(bid) {
+    return !!bid.params.uid;
+  },
+
+  buildRequests: function(validBidRequests) {
+    const auids = [];
+    const bidsMap = {};
+    const bids = validBidRequests || [];
+    let priceType = 'net';
+    let reqId;
+
+    bids.forEach(bid => {
+      if (bid.params.priceType === 'gross') {
+        priceType = 'gross';
+      }
+      if (!bidsMap[bid.params.uid]) {
+        bidsMap[bid.params.uid] = [bid];
+        auids.push(bid.params.uid);
+      } else {
+        bidsMap[bid.params.uid].push(bid);
+      }
+      reqId = bid.bidderRequestId;
+    });
+
+    const payload = {
+      u: utils.getTopWindowUrl(),
+      pt: priceType,
+      auids: auids.join(','),
+      r: reqId,
+    };
+
+    return {
+      method: 'GET',
+      url: ENDPOINT_URL,
+      data: payload,
+      bidsMap: bidsMap,
+    };
+  },
+
+  interpretResponse: function(serverResponse, bidRequest) {
+    serverResponse = serverResponse && serverResponse.body
+    const bidResponses = [];
+    const bidsMap = bidRequest.bidsMap;
+    const priceType = bidRequest.data.pt;
+
+    let errorMessage;
+
+    if (!serverResponse) errorMessage = LOG_ERROR_MESS.emptyResponse;
+    else if (serverResponse.seatbid && !serverResponse.seatbid.length) {
+      errorMessage = LOG_ERROR_MESS.hasEmptySeatbidArray;
+    }
+
+    if (!errorMessage && serverResponse.seatbid) {
+      serverResponse.seatbid.forEach(respItem => {
+        _addBidResponse(_getBidFromResponse(respItem), bidsMap, priceType, bidResponses);
+      });
+    }
+    if (errorMessage) utils.logError(errorMessage);
+    return bidResponses;
+  },
+
+  getUserSyncs: function(syncOptions) {
+    if (syncOptions.pixelEnabled) {
+      return [{
+        type: 'image',
+        url: ADAPTER_SYNC_URL
+      }];
+    }
+  }
+}
+
+function _getBidFromResponse(respItem) {
+  if (!respItem) {
+    utils.logError(LOG_ERROR_MESS.emptySeatbid);
+  } else if (!respItem.bid) {
+    utils.logError(LOG_ERROR_MESS.hasNoArrayOfBids + JSON.stringify(respItem));
+  } else if (!respItem.bid[0]) {
+    utils.logError(LOG_ERROR_MESS.noBid);
+  }
+  return respItem && respItem.bid && respItem.bid[0];
+}
+
+function _addBidResponse(serverBid, bidsMap, priceType, bidResponses) {
+  if (!serverBid) return;
+  let errorMessage;
+  if (!serverBid.auid) errorMessage = LOG_ERROR_MESS.noAuid + JSON.stringify(serverBid);
+  if (!serverBid.adm) errorMessage = LOG_ERROR_MESS.noAdm + JSON.stringify(serverBid);
+  else {
+    const awaitingBids = bidsMap[serverBid.auid];
+    if (awaitingBids) {
+      awaitingBids.forEach(bid => {
+        const bidResponse = {
+          requestId: bid.bidId, // bid.bidderRequestId,
+          bidderCode: spec.code,
```

Comment:
`bidderCode` is set automatically by `bidderFactory` now, setting it here will break aliasing, line and corresponding test lines should be dropped

Reply:
We fixed it, please review.